docs/guides/pagination/overview.md

```diff
@@ -0,0 +1,142 @@
+---
+title: Pagination - Overview
+---
+
+# Pagination
+
+Whenever we deal with lists in GraphQL, we usually need to limit the number of items returned. Surely, we don't want to send massive lists of
+items that take a considerable toll on the server! The goal of this guide is to help you get going fast with pagination!
+
+## Pagination at a glance
+
+We have always dealt with pagination in different situations. Let us take a look at some of the common ways pagination
+can be implemented today!
+
+### Offset based pagination
+
+This pagination style is similar to the syntax we use when looking up database records. Here, the client specifies the number of result to be
+obtained at a time, along with an offset- which usually denotes the number of results to be skipped from the beginning. This type of pagination
+is widely used. Implementing offset-based pagination with an SQL database is straight-forward:
+
+- We count all of the results to determine the total number of pages
+- We use the limit and offset values given to query for the items in the requested page.
+
+Offset based pagination also provides us the ability to jump to a specific page in a dataset.
+
+Let us understand offset based pagination better, with an example. Let us assume that we want to request a list of users, 2 at a time, from a server.
+We start out be sending a request to the server, with the desired limit and offset values.
+
+```json
+{
+  "limit": 2,
+  "offset": 0
+}
+```
```

Comment:
How does the request know we're asking for `users`? Do we want to add a bit more context to this snippet?

Reply:
Actually, is this even supposed to be an actual GraphQL request? We may want to add something short that says, "Don't worry about the actual request format for now. Think about it conceptually, and we'll go into more detail on the actual implementation later"?

Reply:
No, this isnt graphql. It's just an intro to how pagination works. It's just a simple JSON response for conceptual understanding

Reply:
Yes, we could add that bit there.